### PR TITLE
Fixed #4836 & #4840 Added the functionality of cancel upload and also solved the small bug of pausing upload

### LINKED ISSUE #4836: Cancel upload
### What is the user problem or growth opportunity you want to see solved?

Sometimes I select a picture by mistake, or forget to remove EXIF before uploading.

Currently the only option is to reset the app's data, which:
- is inconvenient,
- forces the user to log in again,
- redownloads hundreds of megabytes of contributions,
- loses bookmarks and history.

### How do you know that this problem exists today? Why is this important?

_No response_

### Who will benefit from it?

_No response_

### Anything else you would like to add?

_No response_

### PR DESCRIPTION
**Pull request by arinmodi**
Added the functionality of cancel upload and also solved the small bug of pausing upload

Fixes https://github.com/commons-app/apps-android-commons/issues/4836 & https://github.com/commons-app/apps-android-commons/issues/4840

What changes did you make and why?

I made the cancel button visible after the pause and also added confirmation dialog

For second issue to solve I change the id with string in setTag function of pauseResume button.

Tests performed

Tested build variant on ASUS_X00TD with API level 28

### CODE REVIEW

**Comment by arinmodi** [comment]
Hey @nicolas-raoul, is this working ? Or i have to change something ?

**Comment by nicolas-raoul** [comment]
I tested it yesterday and thought it was not working.
But when I tested it again today, I realized that I was just not waiting for long enough.

Yesterday I thought that Pause was not working on the first tap, so I was always tapping it several times.

Since pausing sometimes takes several seconds, how about showing a modal "Pausing..." popup so that people like me do not tap the button twice? Thanks a lot!

**Comment by arinmodi** [comment]
Ok, will push a changes

**Comment by arinmodi** [comment]
@nicolas-raoul, I tried to display the progress dialog but still the user have to wait for few seconds cause the progress bar still doing it's process even after the pausing, should i fix that progress bar in this PR ?

**Comment by nicolas-raoul** [comment]
Can you hide the "Pausing..." popup just when the "Pause" button disappears?

**Comment by arinmodi** [comment]
pause button disappears or appears ?

**Comment by arinmodi** [comment]
I think we should put a delay of 2 second, so that before closing pausing dialog,

what do you think ?

**Comment by nicolas-raoul** [comment]
I don't think introducing delays is a good idea, as it might be too short or too long depending on the bandwidth and other factors.

**Comment by arinmodi** [comment]
@nicolas-raoul, added the pausing pop up as you said..

but why this progress bar is showing it's progress for few seconds after the pausing it, 

i think it may be cause of the uploadWorker ?

what do you think ?

**Comment by nicolas-raoul** [comment]
Yes, possibly. Making it stop immediately might be a solution, but may be more complex than just monitoring the worker.

**Comment by arinmodi** [comment]
So, what changes required in this ?

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
Please showing a modal "Pausing..." popup that closes exactly when the "Pause" button disappears.
Thanks!

**Comment by arinmodi** [comment]
@nicolas-raoul, I think doing that, user will not be able to see the pop up

**Comment by arinmodi** [comment]
@nicolas-raoul, done requested changes

**Comment by nicolas-raoul** [comment]
Strangely even with your new commit I am not seeing the popup.

**Comment by arinmodi** [comment]
Yes, that's what i am trying to say, 

I think this heppen cause of the, setup of the upload Worker, cause the pause button just changing the state of the button, the actual pause is done by uploadWorker, which takes some time, and also before uploadWorker complete it's task, the pauseUpload completing it's task and returning which eventually calls the closing popup,

In short, currently we are displaying the popup for just setting up the global variable and state, which don't require much time

**Comment by arinmodi** [comment]
So, what do you think about this ?

**Comment by arinmodi** [comment]
Hey @nicolas-raoul, finally it's working now,

You can check test now, it's working better than ever 

Thanks...

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java
Comment (madhurgupta10) [inline app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java:59]: Progress Dialog is deprecated, I suggest using the progress bar.

Comment (arinmodi) [reply app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java:59]: So, I have to create a custom dialog with the progress bar, right ?

Comment (madhurgupta10) [reply app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java:59]: Looking at the scope of the PR, if it's too much trouble to use ProgressBar then I guess ProgressDialog would be fine since there are other places we use it. There is already an issue for it - https://github.com/commons-app/apps-android-commons/issues/3487

@arinmodi Let us know if it possible to use Progress Bar for this PR :)

Comment (arinmodi) [reply app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java:59]: I can make the progress dialog using alert dialog but in that case i also need to create another xml file, is it ok ?

**Comment by arinmodi** [comment]
Hey @madhurgupta10, successfully replaced progress Dialog with the progress Bar and I build it using the alert dialog.

Can you  please review now, 

Thanks.

**Comment by arinmodi** [comment]
Hey @nicolas-raoul, can you please review this

**Comment by arinmodi** [comment]
@nicolas-raoul, is anything needs to change ?

File: app/src/main/java/fr/free/nrw/commons/CommonsApplication.java
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/CommonsApplication.java:139]: Note for the future: Thanks for fixing these! Please do not hesitate to make such changes in a separate pull request, so that this pull request does not contain unrelated changes :-)

By the way:
In memory -> In-memory
contribution -> contributions

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionsListFragment.java
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/contributions/ContributionsListFragment.java:400]: `// Do nothing.`

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java:70]: `// Set a dialog indicating that the upload is being paused. This is needed because pausing an upload might take a dozen seconds.`

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java
Comment (arinmodi) [reply app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java:272]: Hey, if we just pass the string by it's id in setTag, it will set the id of resource like 215678942, like that, not the value of the resource, that's why i am passing like this

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionsListFragment.java
Comment (arinmodi) [reply app/src/main/java/fr/free/nrw/commons/contributions/ContributionsListFragment.java:392]: this string is the title of dialog util, And I forgot to do localization

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java
Comment (arinmodi) [reply app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java:272]: And I do this to solve the issue no. #4840

Comment (arinmodi) [reply app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java:272]: @nicolas-raoul, should I change this ?

**Comment by arinmodi** [comment]
Added the changes

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java
Comment (arinmodi) [reply app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java:273]: changing the text to resource create an issue

**Comment by arinmodi** [comment]
Hey @madhurgupta10, I am blocked to upload the picture, so how can I work on the bugs of the upload feature

**Comment by madhurgupta10** [comment]
> Hey @madhurgupta10, I am blocked to upload the picture, so how can I work on the bugs of the upload feature

I think you can make an appeal to unblock your account, @nicolas-raoul could you please share more information on this?

**Comment by nicolas-raoul** [comment]
@arinmodi Are you blocked on beta or prod? For both you can appeal indeed.

**Comment by arinmodi** [comment]
Hey @nicolas-raoul, is anything else requires ?

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java
Comment (nicolas-raoul) [reply app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java:273]: @4D17Y4 By any chance, would you have any advice on this?

**Comment by arinmodi** [comment]
@nicolas-raoul, changed a bit, please review this....

**Comment by arinmodi** [comment]
@nicolas-raoul, how can i appeal for unblock my account in beta version

**Comment by nicolas-raoul** [comment]
What is your username? "Anonymous me" does not seem to be blocked on beta: https://commons.wikimedia.beta.wmflabs.org/wiki/Special:Log?type=block&user=&page=User%3AAnonymous+me&wpdate=&tagfilter=&subtype=

**Comment by arinmodi** [comment]
my user name Arin Modi

**Comment by nicolas-raoul** [comment]
Apparently you uploaded a meme. In the future, please only upload pictures you have taken yourself (even pictures of a spoon) :-) Thanks! I will try to get you unblocked.

**Comment by arinmodi** [comment]
ok, I just uploaded by mistake

**Comment by arinmodi** [comment]
Hey @nicolas-raoul, any updates on this PR ?

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java
Comment (4D17Y4) [reply app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java]: thanks for updating the values @arinmodi

Comment (arinmodi) [reply app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java]: @4D17Y4, is this okay now ?

**Comment by arinmodi** [comment]
Hey @nicolas-raoul, please review this

**Review by nicolas-raoul — APPROVED** [review]

**Pull request merged**
